packages/edit-site/src/components/global-styles/use-global-styles-output.js

```diff
@@ -283,7 +283,14 @@ export const getNodesWithStyles = ( tree, blockSelectors ) => {
 					styles: value,
 					selector: blockSelectors[ blockName ].selector
 						.split( ',' )
-						.map( ( sel ) => sel + ' ' + ELEMENTS[ elementName ] )
+						.map( ( sel ) => {
+							const elementSelectors =
+								ELEMENTS[ elementName ].split( ',' );
+							return elementSelectors.map(
+								( elementSelector ) =>
+									sel + ' ' + elementSelector
+							);
+						} )
```

Comment:
I've done something similar in https://github.com/WordPress/gutenberg/pull/41786 in order to support pseudo selectors. I'm looking forward to the rebase 😓